### PR TITLE
feat: add event feedback

### LINKED ISSUE #1591: Add Event Feedbacks
**Describe the feature you'd like**
Add Event Feedback which shows a list of all the events feedback in a recycler view once the event/session has been concluded

- [x] Show list of feedbacks
- [ ] Create a feedback for logged-in user
- [ ] Update/Delete a feedback

**Would you like to work on the issue?**
- [x] Yes

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
Fixes #1591 part 1

Changes: added event feedback section for a particular event

Screenshots for the change:
screenshot-1555230073908

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@liveHarshit @iamareebjamal review

**Comment by aggarwalpulkit596** [comment]
@nikit19 couldn't add a screenshot as  there was no Event with a feedback and i was unable to create a feedback manually using the api.I already have created an issue on server for the same

**Comment by aggarwalpulkit596** [comment]
@nikit19 updated

**Comment by aggarwalpulkit596** [comment]
@liveHarshit please review

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsViewModel.kt [hunk 1]
```diff
@@ -52,7 +67,7 @@ class EventDetailsViewModel(private val eventService: EventService, private val
         // location handling
         val BASE_URL = "https://api.mapbox.com/v4/mapbox.emerald/pin-l-marker+673ab7"
         val LOCATION = "(" + event.longitude + "," + event.latitude + ")/" + event.longitude + "," + event.latitude
-        return BASE_URL + LOCATION + ",15/900x500.png?access_token=" + MAPBOX_KEY
+        return "$BASE_URL$LOCATION,15/900x500.png?access_token=$MAPBOX_KEY"
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsViewModel.kt]: Unrelated changes

**Comment by aggarwalpulkit596** [comment]
@liveHarshit @iamareebjamal updated

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal anything left out?

**Comment by iamareebjamal** [comment]
PRF

**Comment by aggarwalpulkit596** [comment]
@iamareebjamal i'm sorry PRF means?

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 2]
```diff
@@ -128,6 +137,15 @@ class EventDetailsFragment : Fragment() {
                 showEventErrorScreen(true)
             })
 
+        eventViewModel.eventFeedback.observe(viewLifecycleOwner, Observer {
+            feedbackAdapter.addAll(it)
+            if (it.isEmpty()) {
+                rootView.feedbackContainer.visibility = View.GONE
```
Comment (addiegupta) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: Replace with `rootView.feedbackContainer.isVisible = false`

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: > Replace with `rootView.feedbackContainer.isVisible = false`

What is the advantage?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: Much clearer and using booleans (primitives)

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: @liveHarshit any other change ?

**Comment by aggarwalpulkit596** [comment]
@liveHarshit updated

File: app/src/main/java/org/fossasia/openevent/general/event/feedback/Feedback.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import com.github.jasminb.jsonapi.annotations.Id
+import com.github.jasminb.jsonapi.annotations.Type
+
+@Type("feedback")
+data class Feedback(
+    @Id(LongIdHandler::class)
+    val id: Long,
+    val rating: String?,
+    val comment: String?,
+    @JsonProperty("deleted-at")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/feedback/Feedback.kt]: Unneeded

Comment (aggarwalpulkit596) [reply app/src/main/java/org/fossasia/openevent/general/event/feedback/Feedback.kt]: @iamareebjamal updated
